Src/AutoFixture/SpecimenBuilderNodeAdapterCollection.cs

```diff
@@ -113,6 +113,8 @@ private ISpecimenBuilderNode AdaptedBuilderNode
         /// <seealso cref="SpecimenBuilderNodeAdapterCollection(ISpecimenBuilderNode, Func{ISpecimenBuilderNode, bool})" />
         public int IndexOf(ISpecimenBuilder item)
         {
+            if (item == null) throw new ArgumentNullException(nameof(item));
```

Comment:
Will we need to update some tests for this change?

Reply:
Good catch 👍  I didn't do deep analysis around it. The reason I added this change is that the newest analyzers are failing about not checking the value. So it was not possible to run the build after I updated the analyzers.

I updated all the other places. Don't think it should be covered with unit tests, as in reality I believe it's only being used by our internal code - it's a collection => graph facade we use for tree mutation.

Let me know if you are fine with that.